// Copyright 2019 The Bazel Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// This file contains the protocol used to communicate between the Bazel client
// and the server. At a high level clients may call the CommandServer.run rpc
// to initiates a Bazel command and CommandServer.cancel to cancel an in-flight
// command. CommandServer.ping may be used to check for server liveness without
// executing any commands. See documentation of individual messages for more
// details.
syntax = "proto3";

package include_processor;

option java_package = "com.google.devtools.build.lib.remote";
option java_multiple_files = true;
option java_outer_classname = "IncludeProcessorProto";

message ProcessIncludesRequest {
  // Current working directory.
  string cwd = 1;

  // Command arguments.
  repeated string args = 2;
}

message ProcessIncludesResponse {
  repeated string includes = 1;
  int32 time_msec = 2;
}

service ProcessIncludesService {
  // Get dependencies of a command from a remote service.
  rpc ProcessIncludes (ProcessIncludesRequest) returns (ProcessIncludesResponse) {}
}